data/transactions/logic/evalCrypto_test.go

```diff
@@ -339,6 +340,112 @@ ecdsa_verify Secp256k1`, hex.EncodeToString(r), hex.EncodeToString(s), hex.Encod
 	require.True(t, pass)
 }
 
+func TestEcdsaWithSecp256r1(t *testing.T) {
+	partitiontest.PartitionTest(t)
+	t.Parallel()
+
+	key, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
+	require.NoError(t, err)
+	pk := elliptic.MarshalCompressed(elliptic.P256(), key.X, key.Y)
+	x := keyToByte(t, key.PublicKey.X)
+	y := keyToByte(t, key.PublicKey.Y)
+
+	// ecdsa decompress tests
+	source := `
+byte 0x%s
+ecdsa_pk_decompress Secp256r1
+store 0
+byte 0x%s
+==
+load 0
+byte 0x%s
+==
+&&`
+	pkTampered1 := make([]byte, len(pk))
+	copy(pkTampered1, pk)
+	pkTampered1[0] = 0                     // first byte is a prefix of either 0x02 or 0x03
+	pkTampered2 := make([]byte, len(pk)-1) // must be 33 bytes length
+	copy(pkTampered2, pk)
+
+	var decompressTests = []struct {
+		key  []byte
+		pass bool
+	}{
+		{pk, true},
+		{pkTampered1, false},
+		{pkTampered2, false},
+	}
+	for i, test := range decompressTests {
+		t.Run(fmt.Sprintf("decompress/pass=%v", test.pass), func(t *testing.T) {
+			t.Log("decompressTests i", i)
+			src := fmt.Sprintf(source, hex.EncodeToString(test.key), hex.EncodeToString(x), hex.EncodeToString(y))
+			if test.pass {
+				testAcceptsWithField(t, src, 5, 6)
+			} else {
+				testPanicsWithField(t, src, 5, 6)
+			}
+		})
+	}
+
+	// ecdsa verify tests
+	source = `
+byte "%s"
+sha512_256
+byte 0x%s
+byte 0x%s
+byte 0x%s
+byte 0x%s
+ecdsa_verify Secp256r1
+`
+	data := []byte("testdata")
+	msg := sha512.Sum512_256(data)
+
+	ri, si, err := ecdsa.Sign(rand.Reader, key, msg[:])
+	require.NoError(t, err)
+	r := ri.Bytes()
+	s := si.Bytes()
+
+	rTampered := make([]byte, len(r))
+	copy(rTampered, r)
+	rTampered[0] += byte(1) // intentional overflow
+
+	var verifyTests = []struct {
+		data string
+		r    []byte
+		pass bool
+	}{
+		{"testdata", r, true},
+		{"testdata", rTampered, false},
+		{"testdata1", r, false},
+	}
+	for _, test := range verifyTests {
+		t.Run(fmt.Sprintf("verify/pass=%v", test.pass), func(t *testing.T) {
+			src := fmt.Sprintf(source, test.data, hex.EncodeToString(test.r), hex.EncodeToString(s), hex.EncodeToString(x), hex.EncodeToString(y))
+			if test.pass {
+				testAcceptsWithField(t, src, 5, 6)
+			} else {
+				testRejectsWithField(t, src, 5, 6)
+			}
+		})
+	}
+
+	// sample sequencing: decompress + verify
+	source = fmt.Sprintf(`#pragma version 6
```

Comment:
I recommend adding assertions to the effect that this errors for `version < fidoVersion` which includes version 6. You can keep the original test cases running and passing locally for `LogicVersion == 7`, but it would be skipped on Circle since `LogicVersion == 6`

Some examples of how I did this for:
* [json_ref](https://github.com/algorand/go-algorand/pull/3514/files#diff-d6e7252ce42ce19fa92f80dd475dff3e611cb2f5c2876395cc851245cd018e6bR5310)
* [bas64_decode](https://github.com/algorand/go-algorand/pull/3514/files#diff-d6e7252ce42ce19fa92f80dd475dff3e611cb2f5c2876395cc851245cd018e6bR5079)